File: php/features.php
<?php
declare(strict_types=1);

// WebPageToMarkdown SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WebPageToMarkdownFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WebPageToMarkdownBaseFeature();
            case "test":
                return new WebPageToMarkdownTestFeature();
            default:
                return new WebPageToMarkdownBaseFeature();
        }
    }
}

File: py/features.py
# WebPageToMarkdown SDK feature factory

from feature.base_feature import WebPageToMarkdownBaseFeature
from feature.test_feature import WebPageToMarkdownTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WebPageToMarkdownBaseFeature(),
        "test": lambda: WebPageToMarkdownTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
